### PR TITLE
18306 investigate possibility to create dn surge sh canary deployments

### PR DESCRIPTION
**Pull request by dqunbp**
https://kontur.fibery.io/Tasks/Task/Investigate-possibility-to-create-DN-surge.sh-canary-deployments-18306

## Summary by CodeRabbit

- **New Features**
  - Introduced automatic deployment of preview environments for pull requests targeting the `main` branch, enhancing testing and previewing of changes before merging.
  
- **Style**
  - Improved readability by removing a blank line in the pre-push script.

### CODE REVIEW

File: .github/workflows/pr-preview.yml [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        run: |
+          sed -i "s|https://disaster.ninja|https://${PREVIEW_API_DOMAIN}|" configs/config.default.json
+          sed -i "s|\"KEYCLOAK_URL\": \"https://keycloak01.kontur.io\"|\"KEYCLOAK_URL\": \"${KEYCLOAK_URL}\"|" configs/config.default.json
+          sed -i "s|\"KEYCLOAK_REALM\": \"kontur\"|\"KEYCLOAK_REALM\": \"${KEYCLOAK_REALM}\"|" configs/config.default.json
+
+      - name: Update .env.production
+        run: |
+          sed -i "s|VITE_BASE_PATH=/active/|VITE_BASE_PATH=/|" .env.production
+          sed -i "s|VITE_STATIC_PATH=static/|VITE_STATIC_PATH=|" .env.production
+
```
[the remark this answers is not in the document]
Comment (amurKontur) [reply .github/workflows/pr-preview.yml:57]: @dqunbp what do you think?

File: .github/workflows/pr-preview.yml [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+        run: cp dist/index.html dist/200.html
+
+      - name: Deploy to surge.sh
+        if: ${{ !cancelled() }}
+        uses: dswistowski/surge-sh-action@v1
+        with:
+          project: ./dist
+          domain: ${{ env.DOMAIN }}
+          login: ${{secrets.SURGE_EMAIL}}
+          token: ${{secrets.SURGE_TOKEN}}
```
Comment (amurKontur) [inline .github/workflows/pr-preview.yml:71]: @dqunbp sure the same as for autotests domain token will work ?

File: .github/workflows/pr-preview.yml [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+    branches:
+      - main
+
+env:
+  # Create a host name for the environment: mr-preview-<PR_NUMBER>.konturio.surge.sh
+  DOMAIN: https://mr-preview-${{ github.event.number }}-konturio.surge.sh
+
+jobs:
+  deploy:
+    runs-on: ubuntu-24.04
```
Comment (amurKontur) [inline .github/workflows/pr-preview.yml]: why not ubuntu-latest?

File: .github/workflows/pr-preview.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (dqunbp) [reply .github/workflows/pr-preview.yml:57]: This is wrong suggestion, I need to do replacements within `.env.production`

File: .github/workflows/pr-preview.yml (hunk 2 above)
Comment (dqunbp) [reply .github/workflows/pr-preview.yml:71]: Yes, it works, domain defined [here](https://github.com/konturio/disaster-ninja-fe/pull/794/files#diff-737d2247cd4b3b6578d2c5413b555a5ca90f6aaa32613770ffdc146900d0469bR10)

**Review by amurKontur — APPROVED** [review]

**Pull request merged**